[WDB-9] Video information

AArlauskas opened the pull request:
PLEASE DELETE YOUR DATABASE TABLES BEFORE STARTING.

1.Added backend logic for parsing information of a video onUpload
2. New Dto for returning full information about video
3. Frontend information drawer
4. Connecting frontend and backend for Video information
5. Fixing a few bugs here and there

AArlauskas committed:
[WDB-118]

AArlauskas committed:
Merge branch 'master' into WDB-9/Video-information

AArlauskas committed:
[WDB-117]

AArlauskas committed:
[WDB-133]

AArlauskas committed:
[WDB-136]

AArlauskas committed:
exporting helper function to utils

AArlauskas committed:
Removed email password

tmaciulis22 commented on frontend/src/components/InformationDrawer/InformationDrawer.js:
```diff
@@ -0,0 +1,110 @@
+  const renderSidebarTitle = () => (
+    <ListItem>
+      <ListItemIcon
+        style={title.length === 0 ? null : { cursor: 'pointer' }}
```
Why are you changing the style here? If you want to make it impossible to close the drawer if title is empty, then why not just disable the icon? if `ListItemIcon` does not have disabled property, then use `IconButton` here(as a child component of `ListItemIcon` or completely replace it).

AArlauskas replied:
Because even with disabled it remains a cursor pointer

tmaciulis22 commented on frontend/src/components/InformationDrawer/InformationDrawer.js:
```diff
@@ -0,0 +1,110 @@
+  return (
+    <SwipeableDrawer
+      anchor="right"
+      open={open || title.length === 0}
```
Why are you checking for title length here?

AArlauskas replied:
This is to make sure that user does not leave a blank title, because this is unhandled in our case

tmaciulis22 commented on frontend/src/components/InformationDrawer/InformationDrawer.js:
```diff
@@ -0,0 +1,110 @@
+      onOpen={onOpen}
+      onClose={onClose}
+    >
+      <List style={{ width: '40vh', height: '100%', position: 'relative' }}>
```
I might be wrong, but I don't think you need `position: 'relative'` here, nor the `height: '100%'`. If you are just blindly copying from `NavDrawer` then don't do it, think critically. In `NavDrawer` these properties were set to have the space taken label on the bottom, but here you don't really need anything to be shown on the bottom of the drawer.

tmaciulis22 commented on frontend/src/containers/PlayerPage/PlayerPage.js:
```diff
@@ -197,22 +226,40 @@ class PlayerPage extends React.Component {
+              open={showInformationDrawer}
+              onOpen={this.toggleInformationDrawer}
+              onClose={this.toggleInformationDrawer}
+              videoTitle={video.title}
+              videoDuration={video.duration}
+              videoSize={video.size}
+              videoFormat={video.format}
+              videoResolution={video.resolution}
```
You don't really have to change anything here if you don't want to, but just passing a whole `video` object would reduce the number of lines here

AArlauskas replied:
I thought this was a bit more declarative making this more of a dummy component, but sure

kibartas commented on backend/DataAccess/Registry/ServiceCollectionExtensions.cs:
```diff
@@ -11,7 +11,7 @@ public static class ServiceCollectionExtensions
         {
             return services.AddDbContext<BackendContext>(options =>
-                options.UseSqlServer(configuration.GetConnectionString(nameof(BackendContext))));
+                options.UseSqlServer(configuration.GetConnectionString(nameof(BackendContext)), b => b.MigrationsAssembly("RestAPI")));
```
Could you explain this change

AArlauskas replied:
Yes. This is because we changed backend structure. This line tells which of the projects is the one to contain the migration assembly. It is necessary to make migrations work

tmaciulis22 reviewed (changes requested):
Overall good job, you are almost done here, but some minor things to look at 🔍

AArlauskas committed:
PR changes

AArlauskas committed:
Merge remote-tracking branch 'origin/master' into WDB-9/Video-information

Commit:
[WDB-9]
add catches to places where there was only then in LibraryPage and add error snackbar when it's not possible to retrieve user videos

Commit:
Merge branch 'WDB-9/Video-information' of github.com:kibartas/PSK-Front-End into WDB-9/Video-information

kibartas reviewed (approved):
Fabulouos
